src/connection.rs

```diff
@@ -1983,259 +1915,349 @@ where
     }
 }
 
-// return true if persistent
+// read request body and prepare outgoing zmq message
 #[allow(clippy::too_many_arguments)]
-async fn server_req_handler<S: AsyncRead + AsyncWrite>(
+async fn server_req_read_body<R: AsyncRead, W: AsyncWrite>(
```

Comment:
can we refactor this into a dedicated function to handle upgrades? to improve readability ...

Reply:
do you mean websocket upgrades? they are rejected by the req mode handler, so there is not much handling going on here

Reply:
went ahead and merged. stream mode will come next, where websocket upgrades are actually accepted, and I'll see about splitting up the code there as applicable